tests/test_generic.py

```diff
@@ -638,3 +648,52 @@ def test_fill_corrupt_data(tmp_path: Path) -> None:
     orig = mikeio.read(infile)
     extracted = mikeio.read(fp)
     assert extracted.n_timesteps == orig.n_timesteps
+
+
+def test_change_datatype_dfs0(tmp_path: Path) -> None:
+    infilename = "tests/testdata/random.dfs0"
+    outfilename = str(tmp_path / "random_datatype107.dfs0")
+    OUT_DATA_TYPE = 107
+
+    change_datatype(infilename, outfilename, datatype=OUT_DATA_TYPE)
+    dfs_out = DfsFileFactory.DfsGenericOpen(outfilename)
+    dfs_in = DfsFileFactory.DfsGenericOpen(infilename)
+
+    n_timesteps_in = dfs_in.FileInfo.TimeAxis.NumberOfTimeSteps
+    n_timesteps_out = dfs_out.FileInfo.TimeAxis.NumberOfTimeSteps
+    datatype_out = dfs_out.FileInfo.DataType
+
+    dfs_out.Close()
+    dfs_in.Close()
+
+    assert datatype_out == OUT_DATA_TYPE
+    assert n_timesteps_in == n_timesteps_out
+    # Also check that data is not modified
+    org = mikeio.read(infilename).to_numpy()
+    new = mikeio.read(outfilename).to_numpy()
+    assert np.allclose(org, new, rtol=1e-08, atol=1e-10, equal_nan=True)
+
+
+def test_change_datatype_dfsu(tmp_path: Path) -> None:
```

Comment:
I don't think it is a good idea to change the data type on dfsu, Mike Zero will refuse to open it. It is good enough with a test for dfs0.